test/langtools/tools/javac/processing/ModuleImportProcessingTest.java

```diff
@@ -0,0 +1,107 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+package tools.javac.annotations;
+
+import toolbox.*;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.TypeElement;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Set;
+
+/**
+ * @test
+ * @bug 8332497
+ * @summary error: javac crashes when annotation processing runs on program with module imports
+ * @library /tools/lib
+ * @modules jdk.compiler/com.sun.tools.javac.api
+ * jdk.compiler/com.sun.tools.javac.main
+ * @build toolbox.JavacTask toolbox.ToolBox toolbox.Assert toolbox.Task
+ * @run main tools.javac.annotations.ModuleImportProcessingTest
+ */
+public class ModuleImportProcessingTest {
+    final toolbox.ToolBox tb = new ToolBox();
+    final Path base = Paths.get(".");
+    final String processedSource = """
+            import module java.base;
+            import java.util.List;
+            import java.lang.annotation.ElementType;
+            import java.lang.annotation.Retention;
+            import java.lang.annotation.RetentionPolicy;
+            import java.lang.annotation.Target;
+            public class Main {
+              public static void main(String[] args) {
+                List.of();
+              }
+              @Ann
+              private void test() {
+                List.of();
+              }
+              @Retention(RetentionPolicy.RUNTIME)
+              @Target(ElementType.METHOD)
+              public @interface Ann {
+              }
+            }
+            """;
+
+    public static void main(String[] args) throws Exception {
+        var t = new ModuleImportProcessingTest();
+        t.test();
+    }
+
+    public void test() throws Exception {
+        tb.writeJavaFiles(base, processedSource);
+        new toolbox.JavacTask(tb)
+                .options(
+                        "-processor", AP.class.getName(),
+                        "--enable-preview",
+                        "--source", System.getProperty("java.specification.version")
```

Comment:
looking good, just a minor comment, here we usually use: `Integer.toString(Runtime.version().feature())` to get the source version, and for indentation we use 4 spaces

Reply:
Im not sure about which indentaion you are talking here. You mean I should use whitespaces instead of tabs, or that I should add spaces betweeen some elements of the code?